### ISSUE TITLE
Consul error: CheckID does not have associated TTL

### ISSUE THREAD

**Issue by alexeyzimarev** [AsynkronIT/protoactor-dotnet#442]
We are randomly getting this error on a ten-node cluster, is there anything we can do to get rid of it?

```
Unhandled Exception: System.AggregateException: One or more errors occurred. (Unexpected response, status code InternalServerError: CheckID "service:TripHandling@10.234.2.30:12000" does not have associated TTL) ---> Consul.ConsulRequestException: Unexpected response, status code InternalServerError: CheckID "service:TripHandling@10.234.2.30:12000" does not have associated TTL
   at Consul.PutNothingRequest.Execute(CancellationToken ct)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Proto.Cluster.Consul.ConsulProvider.BlockingUpdateTtl()
   at Proto.Cluster.Consul.ConsulProvider.<UpdateTtl>b__24_0(Object _)
   at System.Threading.Thread.ThreadMain_ParameterizedThreadStart(Object parameter)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
```

**Comment by alexeyzimarev** [comment]
The issue is transient and can be fixed by retrying but the thread fails. I am planning to spend some time on the Consul provider

**Comment by Tomius** [comment]
The usual cause for this error is when the health check doesn't exist anymore. 
For example a badly configured ```DeregisterCriticalServiceAfter``` parameter on a health check has caused the consul service to deregister, while the underlying process is still alive and is trying to TTL update a no longer existent health check. 
Note: the default value for ```DeregisterCriticalServiceAfter``` is 5 minutes. If a process has longer recovery time from the critical health check state than this or the health check has special semantics, for example it can be critical even when the process is still running, it will cause this error. 
Can you please verify (using the consul server's logs maybe) that the service and the health check is still registered into consul when this error happens?

**Comment by jrouaix** [comment]
@alexeyzimarev I had the same issue and yes fixed it by some error catching in a copy a the cluster provider. Will perhaps provide a PR for this one (but I don't really wanna hide this error).

**Comment by jrouaix** [comment]
@Tomius About the `DeregisterCriticalServiceAfter` when I increase it, I have a lot more error in case of cluster topology change (timeout & split brain), Right now it's better in my case to have reregister of twice the TTL only (ttl is 30 seconds)

**Comment by alexeyzimarev** [comment]
Fixed in #488
